### DISCUSSION TITLE
search with param check failed

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15908] [Q&A and General discussion]
hi，I am searching to milvus-server on python client, code as below:
`nprobe=10
embedding_A = xb[:1] # this is a two dimension matrix, searching vector has same dimension with vector inserted to index.
search_params = {"metric_type": "IP","nprobe":nprobe}
results = collection.search(embedding_A, "embedding", search_params, limit=2, expr="", output_fields=["mid"])
`
Error occurred as shown below. What's happening?
RPC error: [search], <BaseException: (code=1, message=no Available QueryNode result, filter reason [UnexpectedError] Assert "adapter->CheckSearch(conf, index_type, field_indexing->indexing_->index_mode())" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/SearchOnSealed.cpp:91
 => [SearchOnSealed]Search params check failed

**Comment by yhmo** [comment]
Seems the same problem with this issue: https://github.com/milvus-io/milvus/issues/15810

Could you show your index parameters? Typically, the create_index() requires a "metric_type":
```
 index = {
     "index_type": "IVF_FLAT",
     "metric_type": "IP",
     "params": {"nlist": 128},
 }
 collection.create_index("xxx", index)
```

**Reply by xiaojuan8** [reply]
ok, thanks, when I drop the collection, and create this collection again, the error disappear.

**Reply by yhmo** [reply]
I guess it is a bug.
If you encounter it again and found how to reproduce it, please tell us.

**Reply by xiaojuan8** [reply]
It is easy to reproduce.
1. create collection
2. insert data
3. create index
4. load collection
5. search query
Then param check failed.

my code is as follows:

from numpy import int64
from pymilvus import (
    connections,
    list_collections,
    FieldSchema, CollectionSchema, DataType,
    Collection, Index
)
import random
from math import sqrt
import os
import json

data_train_input="../../train_embs" #embedding输入
host = '10.30.88.57'
port = '32491'
connections.connect(host=host, port=port)
print(list_collections())

dim = 32
fields = [
    FieldSchema(name="mid", dtype=DataType.INT64, is_primary=True, auto_id=False),
    FieldSchema(name="embedding", dtype=DataType.FLOAT_VECTOR, dim=dim),
]
schema = CollectionSchema(fields, description='desc of collection')
collection_name = "milvus_recall_at_n"
collection = Collection(collection_name, schema, consistency_level="Strong")

list_dirs = os.walk(data_train_input)
xb = []
xi = []
for root, dirs, files in list_dirs:
    for f in files:
        f_name = os.path.join(root, f)
        print("reading %s"%f_name)
        with open(f_name, encoding='utf-8') as fi:
            for line in fi:
                line_vec = line.split(":")
                vec_id = int64(line_vec[0])
                vec = json.loads("[" + line_vec[1] + "]")
                xb.append(vec)
                xi.append(vec_id)

size=len(xb)
max_per_insert=3000000
segment = (size + max_per_insert - 1) / max_per_insert
segment = int(segment)
for j in range(1):
    for i in range(segment):
        begin = i * max_per_insert
        end = min((i + 1) * max_per_insert, size)
        entities = [
            xi[begin:end], # field id
            xb[begin:end],  # field embedding
        ]
        print("round %s insert segment %s"%(j,i))
        insert_result = collection.insert(entities)
        print("round %s insert count : %s"%(i,insert_result.insert_count))

print("collecttion : %s has entities num : %s"%(collection_name,collection.num_entities))

print("creating index...")
#create index
size = len(xb)
alpha=2
nlist=int(alpha * sqrt(size))
params={"nlist":nlist, "m":4}
index_params = {"index_type":"IVF_PQ", "metric_type":"IP", "params":params}
index = collection.create_index("embedding", index_params)

print("loading index...")
#load collection
collection.load()

print("search index...")
#search index
nprobe=10
embedding_A = xb[:1]
search_params = {"metric_type": "IP","nprobe":nprobe}
results = collection.search(embedding_A, "embedding", search_params, limit=2, expr="", output_fields=["mid"])

print("search results:")
print(results)

the data in the file is like this:
22090214:0.172577888,-0.0316003412,0.267329425,-0.141024962,0.136989176,-0.0309890844,0.165340081,0.275654405,-0.237941578,0.0765085369,0.00819389801,0.0237542912,-0.0219437014,-0.263107568,0.12053021,-0.17982237,-0.112910189,-0.0615054928,0.0163544491,-0.262723297,0.296986908,-0.0666301474,-0.062152192,-0.216569304,-0.270462275,0.069062151,-0.262509942,-0.461815566,-0.0436953418,-0.290283889,0.017662622,-0.0100122625:-0.990682125,2.78805423,0.634041607,-3.80680919,-0.849351346,2.06489205,2.40606999,0.954586923,0.320821583,-2.22192883,0.664527357,-1.39065993,-3.15581489,0.942567468,-0.347922206,0.296670675,3.45141125,-1.74645233,2.61467147,-0.693552017,0.154658243,0.719980896,0.827962041,0.0968127623,2.26327682,-0.188252077,1.87235785,2.15982866,3.75250125,-0.838748157,1.89939392,-0.735427439

**Reply by weiwang19** [reply]
Hi, 
I am seeing the same error with vector search: 

RPC error: [search], <BaseException: (code=1, message=no Available QueryNode result, filter reason [UnexpectedError] Assert "adapter->CheckSearch(conf, index_type, field_indexing->indexing_->index_mode())" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/SearchOnSealed.cpp:91
 => [SearchOnSealed]Search params check failed

Index type: IVF_SQ8

Is there a fix for the issue?

**Reply by xiaofan-luan** [reply]
> Hi, I am seeing the same error with vector search:
> 
> RPC error: [search], <BaseException: (code=1, message=no Available QueryNode result, filter reason [UnexpectedError] Assert "adapter->CheckSearch(conf, index_type, field_indexing->indexing_->index_mode())" at /go/src/github.com/milvus-io/milvus/internal/core/src/query/SearchOnSealed.cpp:91 => [SearchOnSealed]Search params check failed
> 
> Index type: IVF_SQ8
> 
> Is there a fix for the issue?

CheckSearch fail means your search parameter is wrong, please check your metric types and other search parameters

**Reply by xiaofan-luan** [reply]
we are working on refine the error message

**Reply by weiwang19** [reply]
My index params are 
vector_index_param = {
        "index_type": "IVF_SQ8",
        "metric_type": "L2",
        "params": {"nlist": 1024},
    }

and search parameter: 
search_params = {"index_type":"IVF_SQ8", "metric_type": "L2","param":{"nlist":1024}}

Can you help to see if there is anything wrong with the params? The error also happens intermittenly when I use the Attu UI for vector search.

**Comment by xiaofan-luan** [comment]
@yhmo please help on checking if we change the error message

**Comment by yhmo** [comment]
@xiaojuan8 
The correct search parameter format should be:

```
search_params = {"metric_type": 'IP', "params": {"nprobe": nprobe}}
results = collection.search(embedding_A, "embedding", search_params, limit=2, expr="", output_fields=["mid"])
```

**Comment by yhmo** [comment]
@weiwang19 
The correct search parameter format should be:
`search_params = {"metric_type": "L2","params":{"nprobe":1024}}`

**Reply by weiwang19** [reply]
@yhmo 
changed to search_params = {"metric_type": "L2","params":{"nlist":1024}} but still seeing the error.

**Reply by yhmo** [reply]
Sorry, should be: `search_params = {"metric_type": "L2","params":{"nprobe":1024}}`

**Comment by yaroslavm** [comment]
It turns out that `nprobe` parameter becomes mandatory after some size (which actually quite small anyway). I put some more details into https://github.com/milvus-io/milvus/issues/15294
